code/controllers/subsystem/job.dm

```diff
@@ -277,8 +277,14 @@ SUBSYSTEM_DEF(job)
 
 	job.after_spawn(new_character, player, joined_late) // note: this happens before new_character has a key!
 
-	return new_character
+	if(!isxenosjob(job))
+		// try to give them their medals
+		var/datum/medal_persistence/medals = get_medal_persistence_for_ckey(player.ckey)
+		ASYNC
+			medals.load_medals_from_db(player)
```

Comment:
/datum/medal_persistence/New(ckey)
	ckey = ckey
	owner = GLOB.player_details[ckey]
	load_medals_from_db()
	
	new also calls it one of these shouldnt be here